### DISCUSSION TITLE
Error: Failed to find Server Action. This request might be from an older or newer deployment. Original error: Cannot read properties of undefined (reading 'workers')

### DISCUSSION THREAD

**Discussion by AndreaDelliGatti** [vercel/next.js#65969] [Help]
### Summary

I'll explain what my problem is. 
I have a package in which I expose a client component that calls a server action on a button click.

Now if I use this component in my nextjs application within a server component it works normally.
However, if I use this component within another client component I receive the following error.

```
Error: Failed to find Server Action. This request might be from an older or newer deployment. Original error: Cannot read properties of undefined (reading 'workers')
```

Help me solve this problem. I can't figure it out.

**Comment by maikrolf** [comment]
i have this issue too can not find anything on this. did you found already a solution?

**Reply by hroomdeveloper** [reply]
same

**Reply by Mnigos** [reply]
same

**Reply by maikrolf** [reply]
you need to add the path of the action function to the first client component before the load, then it will work

**Reply by Mnigos** [reply]
downgrading from `14.2.8` to `14.2.7` fixed this issue for me.

UPDATE: works with `14.2.10` and `14.2.11`

**Reply by andresg747** [reply]
Hey @maikrolf, when you say "add the path of the action function" you refer to importing the action function in the parent client component?
Would you be able to provide a small example? 
Thanks!

**Reply by maikrolf** [reply]
yes import the action file in the client before its called. then it will find the action

"use client"
import { actions } from "./......."

**Comment by khuezy** [comment]
Seems like a bug with importing, eg:

`index.ts`
```typescript
export * from './actions'
```

`actions.ts`
```
'use server'
export async function someAction...
```

If you import from the `index` file, then you get the error in the OP. But if you import directly from the `actions.ts`, then it works.
This didn't error before but now it does.

cc @shuding (sorry for the ping)

**Reply by maikrolf** [reply]
sorry your right, 

you can just use
`import '@/actions`
in my case and have a default export

**Reply by shuding** [reply]
Thanks for flagging. I opened #70317 to address it, will bring up to our team!

**Reply by khuezy** [reply]
Thanks this is fixed in `.162`

**Reply by andresg747** [reply]
Hey @khuezy - I can confirm that this issue still happens using the latest 15.0.0-rc.1 version.
<details>
  <summary>Error Details</summary>
Error: Failed to find Server Action "23232cf808570a27f63e223cde3dd776adc72835". This request might be from an older or newer deployment. Original error: Cannot read properties of undefined (reading 'workers')
    at rj (/var/task/node_modules/.pnpm/next@15.0.0-rc.0_@babel+core@7.24.7_@opentelemetry+api@1.9.0_react-dom@19.0.0-rc-f994737d14-2_niaqmqcs6w5rz3m2tdu2ujwiha/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:128:2550)
    at /var/task/node_modules/.pnpm/next@15.0.0-rc.0_@babel+core@7.24.7_@opentelemetry+api@1.9.0_react-dom@19.0.0-rc-f994737d14-2_niaqmqcs6w5rz3m2tdu2ujwiha/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:128:708
    at AsyncLocalStorage.run (node:async_hooks:346:14)
    at r$ (/var/task/node_modules/.pnpm/next@15.0.0-rc.0_@babel+core@7.24.7_@opentelemetry+api@1.9.0_react-dom@19.0.0-rc-f994737d14-2_niaqmqcs6w5rz3m2tdu2ujwiha/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:127:8138)
    at nn (/var/task/node_modules/.pnpm/next@15.0.0-rc.0_@babel+core@7.24.7_@opentelemetry+api@1.9.0_react-dom@19.0.0-rc-f994737d14-2_niaqmqcs6w5rz3m2tdu2ujwiha/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:129:1146)
    at /var/task/node_modules/.pnpm/next@15.0.0-rc.0_@babel+core@7.24.7_@opentelemetry+api@1.9.0_react-dom@19.0.0-rc-f994737d14-2_niaqmqcs6w5rz3m2tdu2ujwiha/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:130:726
    at AsyncLocalStorage.run (node:async_hooks:346:14)
    at Object.run (/var/task/web/patchedAsyncStorage.cjs:5:76)
    at Object.wrap (/var/task/node_modules/.pnpm/next@15.0.0-rc.0_@babel+core@7.24.7_@opentelemetry+api@1.9.0_react-dom@19.0.0-rc-f994737d14-2_niaqmqcs6w5rz3m2tdu2ujwiha/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:125:19860)
    at /var/task/node_modules/.pnpm/next@15.0.0-rc.0_@babel+core@7.24.7_@opentelemetry+api@1.9.0_react-dom@19.0.0-rc-f994737d14-2_niaqmqcs6w5rz3m2tdu2ujwiha/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:130:616
</details>

**Reply by khuezy** [reply]
@andresg747 maybe you have an old `node_modules` that needs cleaning up? I would delete it an reinstall. I see that the log still shows `rc.0` even though you're on the latest.
I just tested this on the 15.0.0 release and it works. Maybe your setup is different and has an edge case that wasn't fixed?

**Reply by andresg747** [reply]
You're right. Nice catch. 
I'll investigate. Thanks!

**Reply by andresg747** [reply]
@khuezy @shuding  Actually this is the latest occurrence of the error. Confirming we're using v15.0.0-rc.1:
<details>
  <summary>Error details</summary>
Error: Failed to find Server Action "6893e6c3a60c6cb4ba61d722598dbbfaa52fef48". This request might be from an older or newer deployment. Original error: Cannot read properties of undefined (reading 'workers')
    at rd (/var/task/node_modules/.pnpm/next@15.0.0-rc.1_@babel+core@7.24.7_@opentelemetry+api@1.9.0_react-dom@19.0.0-rc-f994737d14-2_mknobfj5guh47wwahl3vqagiyq/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:127:2692)
    at /var/task/node_modules/.pnpm/next@15.0.0-rc.1_@babel+core@7.24.7_@opentelemetry+api@1.9.0_react-dom@19.0.0-rc-f994737d14-2_mknobfj5guh47wwahl3vqagiyq/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:127:757
    at AsyncLocalStorage.run (node:async_hooks:346:14)
    at rc (/var/task/node_modules/.pnpm/next@15.0.0-rc.1_@babel+core@7.24.7_@opentelemetry+api@1.9.0_react-dom@19.0.0-rc-f994737d14-2_mknobfj5guh47wwahl3vqagiyq/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:126:11360)
    at n6 (/var/task/node_modules/.pnpm/next@15.0.0-rc.1_@babel+core@7.24.7_@opentelemetry+api@1.9.0_react-dom@19.0.0-rc-f994737d14-2_mknobfj5guh47wwahl3vqagiyq/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:132:1223)
    at /var/task/node_modules/.pnpm/next@15.0.0-rc.1_@babel+core@7.24.7_@opentelemetry+api@1.9.0_react-dom@19.0.0-rc-f994737d14-2_mknobfj5guh47wwahl3vqagiyq/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:132:6300
    at AsyncLocalStorage.run (node:async_hooks:346:14)
    at tm (/var/task/node_modules/.pnpm/next@15.0.0-rc.1_@babel+core@7.24.7_@opentelemetry+api@1.9.0_react-dom@19.0.0-rc-f994737d14-2_mknobfj5guh47wwahl3vqagiyq/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:124:48121)
    at /var/task/node_modules/.pnpm/next@15.0.0-rc.1_@babel+core@7.24.7_@opentelemetry+api@1.9.0_react-dom@19.0.0-rc-f994737d14-2_mknobfj5guh47wwahl3vqagiyq/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:132:6136
    at AsyncLocalStorage.run (node:async_hooks:346:14)
</details>

**Reply by khuezy** [reply]
Can you try w/ 15.0.0? If it still errors, you'll need to provide a min repro.

**Comment by benjick** [comment]
I get this with 15.1.5

> Error: Failed to find Server Action "7ff9225b349ed9316d9c3fc2a461800836da4fd0a1". This request might be from an older or newer deployment. Original error: Cannot read properties of undefined (reading 'workers')

How can I even track which action it is?

**Reply by maikrolf** [reply]
you need to call your action.ts in the client before the server action is called 

```
"use client"
import "../action/action.ts" 
```

something like this

**Reply by LikeDreamwalker** [reply]
Same at 15.1.2, really weird, and even more this issue won't happen at every request, hard to figure out.

**Comment by mattmar10** [comment]
Same issue here

**Comment by LikeDreamwalker** [comment]
For everyone who is suffering this, you can read https://github.com/vercel/next.js/issues/75541#issue-2824452242

**Reply by benjick** [reply]
That env var hasn't worked for me

**Reply by LikeDreamwalker** [reply]
> That env var hasn't worked for me

It probably won't work for everyone, just a reminder for people who want to get more info about this issue🫡

**Comment by jperezrealini** [comment]
I'm seeing this error constantly on Vercel:

![image](https://github.com/user-attachments/assets/7ac9cbb3-057e-402d-98f3-8d07b0a7793e)

I'm on nextjs version 15.2.3 and have skew protection enabled.

**Reply by faishaldev** [reply]
I'm working on Next.js version 15.1.7 and facing the same issue. Did you solve it?

**Reply by jperezrealini** [reply]
Nope, still showing up. I'm on `15.3.1-canary.10` now.

**Comment by ifielder** [comment]
Having the same issue.

Is there an ETA on this fix?

**Reply by DerekWolfie** [reply]
Still having the same issue
